packages/plugin-defender/src/hardhat/index.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+import '@nomiclabs/hardhat-ethers';
+import { extendConfig, extendEnvironment } from 'hardhat/config';
+import { lazyObject } from 'hardhat/plugins';
+import { HardhatConfig, HardhatUserConfig } from 'hardhat/types';
+import { ProposeUpgradeFunction } from './propose-upgrade';
+import './type-extensions';
+
+export interface HardhatDefenderUpgrades {
+  proposeUpgrade: ProposeUpgradeFunction;
+}
+
+export interface HardhatDefenderConfig {
+  apiKey: string;
+  apiSecret: string;
+}
+
+extendConfig((config: HardhatConfig, userConfig: Readonly<HardhatUserConfig>) => {
+  if (!userConfig.defender) {
+    console.warn(
+      `Defender API key and secret are not set. Please add the following to your hardhat.config.js exported configuration:\n\n{\n  defender: {\n    apiKey: 'YOUR_API_KEY',\n    apiSecret: 'YOUR_API_SECRET'\n  }\n}\n`,
```

Comment:
This can be a little nicer if we use `JSON.stringify(sampleConfig, null, 2)`.

Also, this is a matter of style but I've been avoiding "Please" in this kind of message, jumping straight to the indication.